src/java.base/share/native/libzip/Adler32.c

```diff
@@ -40,7 +45,7 @@ Java_java_util_zip_Adler32_update(JNIEnv *env, jclass cls, jint adler, jint b)
     Bytef buf[1];
 
     buf[0] = (Bytef)b;
-    return adler32(adler, buf, 1);
+    return adler32((unsigned int)adler, buf, 1);
```

Comment:
This makes assumptions about the width of `unsigned int`.
I think a better expression would be `jint_to_jlong(adler) & 0xFFFFFFFFL`, perhaps via a new macro.

Reply:
I'm surprised we don't see warnings about the implicit cast from `uLong` to `jint` in this return statement, nor in the assignment on line 57. Perhaps they need explicit casts.